test/lib/sender/matchers.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sender
+
+import (
+	"errors"
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/event"
+	cetest "github.com/cloudevents/sdk-go/v2/test"
+	cetypes "github.com/cloudevents/sdk-go/v2/types"
+)
+
+func MatchStatusCode(status int) cetest.EventMatcher {
+	return cetest.AllOf(
+		cetest.HasType(EventType),
+		cetest.AnyOf(
+			// Because extensions could lose type information during serialization
+			// (eg when they're transported as http headers) the assert should match or the string or the int
+			cetest.HasExtension(ResponseStatusCodeExtension, cetypes.FormatInteger(int32(status))),
+			cetest.HasExtension(ResponseStatusCodeExtension, status),
+		),
+	)
+}
+
+func MatchInnerEvent(matchers ...cetest.EventMatcher) cetest.EventMatcher {
```

Comment:
Is this function used anywhere?

Reply:
Not yet, but i suppose we'll need it for broker data-plane tests

Reply:
@lberk might want to take a look too ?